### PR TITLE
added a check for zero stride

### PR DESCRIPTION
**Pull request by anjali411**

### CODE REVIEW

**Comment by anjali411** [comment]
Issue: #27598

**Comment by SsnL** [comment]
Can we instead change is_stride_neg to is_stride_nonpos?

**Comment by anjali411** [comment]
@SsnL good point! updated.

**Review by albanD — APPROVED** [review]
LGTM

File: aten/src/ATen/native/Convolution.cpp [hunk 1]
```diff
@@ -381,7 +380,7 @@ static void check_shape_forward(const at::Tensor& input,
 
   TORCH_CHECK(!params.is_padding_neg(), "negative padding is not supported");
   TORCH_CHECK(!params.is_output_padding_neg(), "negative output_padding is not supported");
-  TORCH_CHECK(!params.is_stride_neg(), "negative stride is not supported");
+  TORCH_CHECK(!params.is_stride_nonpos(), "non positive stride is not supported");
```
Comment (yf225) [inline aten/src/ATen/native/Convolution.cpp]: nit: maybe "non-positive" instead of "non positive"?

Comment (anjali411) [reply aten/src/ATen/native/Convolution.cpp]: done

**Review by yf225 — APPROVED** [review]

**Comment by gchanan** [comment]
Some minor feedback on this:
1) It would be a slight improvement to focus on the effect of the change instead of how the change works.  i.e. it's more important that this PR fixes a segfault when the convolution stride == 0 than that it does this by checking.

2) It's a bit concerning that the code in question from the original PR still exists in THNN.  That is, if someone manages to call that code via another path it will still crash.  We should also change the check in THNN.

**Pull request closed without merging**